router/router.go

```diff
@@ -53,21 +55,17 @@ type Router struct {
 	acceptLimiter   *TokenBucket
 }
 
-type PacketSource interface {
-	ReadPacket() ([]byte, error)
-}
+func NewRouter(config Config, name PeerName, nickName string) *Router {
+	router := &Router{Config: config, gossipChannels: make(GossipChannels)}
 
-type PacketSink interface {
-	WritePacket([]byte) error
-}
+	if router.Bridge == nil {
+		router.Bridge = NullBridge{}
+	}
 
-type PacketSourceSink interface {
-	PacketSource
-	PacketSink
-}
+	if router.Overlay == nil {
+		router.Overlay = NullOverlay{}
+	}
```

Comment:
IMO `Config` should be immutable. Perhaps introduce a `NewConfig()` that sets `Bridge` and `Overlay`?